### PR TITLE
Log output to filesystem

### PR DESCRIPTION
**Pull request by sethdusek**
For #5 

Some questions:
- Is rotating log files daily fine or should I adjust the rotation interval?
- Writing logs to ./chaincash.log.d/chaincash.log*. I was looking for where the standard place to put log files on Linux for user programs is, and there doesn't seem to be one (besides /var/log which requires special privileges to create a log directory). So I assume putting this in the current directory is fine.

### CODE REVIEW

**Comment by ross-weir** [comment]
> Is rotating log files daily fine or should I adjust the rotation interval?

yeah that's fine

> Writing logs to ./chaincash.log.d/chaincash.log*. I was looking for where the standard place to put log files on Linux for user programs is, and there doesn't seem to be one (besides /var/log which requires special privileges to create a log directory). So I assume putting this in the current directory is fine.

Lets add `directories-rs` dependency and put logs in the dirs specified here: https://github.com/dirs-dev/directories-rs/issues/81

**Comment by sethdusek** [comment]
> Lets add `directories-rs` dependency and put logs in the dirs specified here: [dirs-dev/directories-rs#81](https://github.com/dirs-dev/directories-rs/issues/81)

Done! Also watching the github issue so if (and when) this is built-in to directories-rs I can replace our implementation with that

File: crates/chaincash_cli/src/main.rs [hunk 1]
```diff
@@ -23,8 +24,26 @@ struct Cli {
...
+    } else if cfg!(target_os = "windows") {
+        Some(base_dirs.data_dir().to_owned())
+    } else {
+        None
+    }
+}
+
 impl Cli {
     pub async fn execute(&self) -> Result<()> {
+        let mut log_directory = get_log_directory().unwrap_or_default();
```
Comment (ross-weir) [inline crates/chaincash_cli/src/main.rs]: Lets use `std::env::current_dir` as fallback default to be explicit

**Review by ross-weir — CHANGES_REQUESTED** [review]

**Comment by ross-weir** [comment]
> > Lets add `directories-rs` dependency and put logs in the dirs specified here: [dirs-dev/directories-rs#81](https://github.com/dirs-dev/directories-rs/issues/81)
> 
> Done! Also watching the github issue so if (and when) this is built-in to directories-rs I can replace our implementation with that

Amazing, thank you!

Just one more small thing and will merge.

Whats your discord/telegram? I'll get kushti to contact you for bounty payment

**Comment by sethdusek** [comment]
> > > Lets add `directories-rs` dependency and put logs in the dirs specified here: [dirs-dev/directories-rs#81](https://github.com/dirs-dev/directories-rs/issues/81)
> > 
> > 
> > Done! Also watching the github issue so if (and when) this is built-in to directories-rs I can replace our implementation with that
> 
> Amazing, thank you!
> 
> Just one more small thing and will merge.
> 
> Whats your discord/telegram? I'll get kushti to contact you for bounty payment

My Discord username is dusek_. Thanks!

**Comment by kushti** [comment]
@SethDusek please provide your Ergo address (ordinary or stealth) over Discord (sent you DM there) or email kushti@protonmail.ch

**Pull request merged**